src/cascadia/TerminalControl/UiaTextRange.cpp

```diff
@@ -45,67 +45,6 @@ std::deque<UiaTextRange*> UiaTextRange::GetSelectionRanges(_In_ IUiaData* pData,
     return ranges;
 }
 
-UiaTextRange* UiaTextRange::Create(_In_ IUiaData* pData,
-                                   _In_ IRawElementProviderSimple* const pProvider)
-{
-    try
-    {
-        return new UiaTextRange(pData, pProvider);

```

Comment:
it looks like, based on this, that the constructors can throw. They _shouldn't_, but it looks like they can. Here's what you should do.

1. Acknowledge that class creation might return a failure
1. Presume that failure should come in the form of an `HRESULT`
1. move all code that might fail -- either by throwing or returning a bad status code -- into a member function called `RuntimeClassInitialize`
1. make it safe:

```cpp
struct CWhatever: public WRL::RuntimeClass<...>
{
    HRESULT RuntimeClassInitialize(Arg1 a, Arg2 b);
};

HRESULT CWhatever RuntimeClassInitialize(Arg1, Arg2)
try
{
    // if something throws, it will be caught;
    RETURN_IF_FAILED(other_thing_with_an_hr);
}
CATCH_RETURN();
```

5. use `CWhatever* instance; RETURN_IF_FAILED(WRL::MakeAndInitialize<CWhatever>(&instance, a1, a2))`.